src/baldr/incident_singleton.h

```diff
@@ -97,24 +65,66 @@ struct incident_singleton_t {
     if (state->signal.wait_until(lock, when) == std::cv_status::timeout) {
       throw std::runtime_error("Unable to initialize incident watcher in the configured time period");
     }
-    // let this run forever
-    watcher.detach();
+  }
+
+  /**
+   * The default interrupt for the watcher threads main loop. Since its not set, the watcher loop
+   * will run forever. This method is provided so the unit tests can exercise the constructor
+   * @return
+   */
+  virtual std::function<bool(size_t)> interrupt() {
+    return {};
+  }
+
+  /**
+   * Read the contents of a file into an incident tile
+   * @param filename   name of the file on the file system to read into memory
+   * @return a shared pointer with the data of the tile or an empty pointer if it could not be read
+   */
+  static std::shared_ptr<const valhalla::IncidentsTile> read_tile(const std::string& filename) {
+    // open the file for reading. its normal for this to fail when the file has been removed
+    std::ifstream file(filename, std::ios::in | std::ios::binary);
+    if (!file.is_open()) {
+      return {};
```

Comment:
Would it be good to log a warning here? I don't think this should occur during normal operations?

Reply:
the comment is maybe not expressive enough. if your incident implementation removes files from the file system it can happen that you go to load the file and its been yanked before you can. or if you are using the memory mapped log, you can reset timestamp and remove the file from the filesystem and this code will be triggered. in either case its normal operation to expect the open to fail. previously i had a warning here but then i realized removing files from the filesystem is a pretty normal thing to do to decomission an incident tile